core/src/test/java/org/apache/iceberg/TestSortOrder.java

```diff
@@ -320,6 +320,9 @@ public void testColumnDropWithSortOrder() {
         "Order ID must match", TableMetadata.INITIAL_SORT_ORDER_ID + 1, actualOrder.orderId());
     Assert.assertEquals(
         "Schema must have one less column", initialColSize - 1, table.schema().columns().size());
+
+    // ensure that the table metadata can be serialized and reloaded with an invalid order
+    TableMetadataParser.fromJson(TableMetadataParser.toJson(table.ops().current()));
```

Comment:
nit: generally speaking I think we should have an assertion here, even if we only want to make sure that code doesn't fail